disnake/appinfo.py

```diff
@@ -402,6 +448,154 @@ def user_install_type_config(self) -> Optional[InstallTypeConfiguration]:
+        description: Optional[str] = MISSING,
+        role_connections_verification_url: Optional[str] = MISSING,
+        install_params: Optional[InstallParams] = MISSING,
+        guild_install_type_config: Optional[InstallTypeConfiguration] = MISSING,
+        user_install_type_config: Optional[InstallTypeConfiguration] = MISSING,
+        flags: ApplicationFlags = MISSING,
+        icon: Optional[AssetBytes] = MISSING,
+        cover_image: Optional[AssetBytes] = MISSING,
+        interactions_endpoint_url: Optional[str] = MISSING,
+        tags: Sequence[str] = MISSING,
+        event_webhooks_url: Optional[str] = MISSING,
+        event_webhooks_status: ApplicationEventWebhookStatus = MISSING,
+        event_webhooks_types: Sequence[str] = MISSING,
+    ) -> AppInfo:
+        """|coro|
+
+        Edit's the application's information.
+
+        All parameters are optional.
+
+        .. versionadded:: |vnext|
+
+        Parameters
+        ----------
+        description: Optional[:class:`str`]
+            The application's description.
+        flags: :class:`ApplicationFlags`
+            The application's public flags.
+
+            This is restricted to only affecting the limited intent flags:
+            :attr:`~ApplicationFlags.gateway_guild_members_limited`,
+            :attr:`~ApplicationFlags.gateway_presence_limited`, and
+            :attr:`~ApplicationFlags.gateway_message_content_limited`.
+
+            .. warning::
+                Disabling an intent that you are currently requesting during your current session
+                will cause you to be disconnected from the gateway. Take caution when providing this parameter.
+
+        tags: List[:class:`str`]
+            The application's tags.
+        install_params: Optional[:class:`InstallParams`]
+            The installation parameters for this application.
+
+            Cannot be provided with ``custom_install_url``.
+        custom_install_url: Optional[:class:`str`]
+            The custom installation url for this application.
+        role_connections_verification_url: Optional[:class:`str`]
+            The application's role connection verification entry point,
+            which when configured will render the app as a verification method
+            in the guild role verification configuration.
+        interactions_endpoint_url: Optional[:class:`str`]
+            The application's interactions endpoint URL.
+        icon: Optional[|resource_type|]
+            Update the application's icon asset, if any.
+        cover_image: Optional[|resource_type|]
+            Update the cover_image for rich presence integrations.
+        guild_install_type_config: Optional[:class:`InstallTypeConfiguration`]
+            The guild installation type configuration for this application.
+        user_install_type_config: Optional[:class:`InstallTypeConfiguration`]
+            The user installation type configuration for this application.
```

Comment:
```suggestion
        guild_install_type_config: Optional[:class:`InstallTypeConfiguration`]
            The guild installation type configuration for this application.

            Cannot be provided with ``custom_install_url``.
        user_install_type_config: Optional[:class:`InstallTypeConfiguration`]
            The user installation type configuration for this application.

            Cannot be provided with ``custom_install_url``.
```

Reply:
It can though. Just without the InstallParams in the class.

Reply:
In order to set a custom install URL and update an app to be only be installed to users, the payload must be an empty object for each install type. 

Quite honestly, I debated adding support for booleans for these variables if custom_url is provided.

Reply:
> It can though. Just without the InstallParams in the class.

ah, right.
I think it's fine to leave as-is, then.